python/ccxt/base/exchange.py

```diff
@@ -688,7 +688,8 @@ def key_exists(dictionary, key):
                 return dictionary[key] is not None
             else:
                 return False
-        if key in dictionary:
+        # check for string type, to avoid  `TypeError: 'in ' requires string as left operand, not int`, i.e. safeString(dict, 0)
+        if isinstance(key, str) and key in dictionary:
```

Comment:
@ttodua Can you please elaborate a bit more on this? I'm not getting that error when using integer keys

Reply:
I had incorrect impression. the linked issue seems to be having different problem - not `integer in list`, but `integer in string` check. The error log was:
```
File 'ccxt/bitfinex.py', line 1495, in handle_errors
    firstChar = self.safe_string(body, 0)
File 'ccxt/base/exchange.py', line 704, in safe_string
    return str(dictionary[key]) if Exchange.key_exists(dictionary, key) else default_value
File 'ccxt/base/exchange.py', line 688, in key_exists
    if key in dictionary:
TypeError: 'in ' requires string as left operand, not int
```

so there `body` seems to be `string` already. how it should be handled effectively ?  that might need to fix globally, not for just `bitfinex`...

Reply:
@ttodua 

> how it should be handled effectively ?

```JavaScript
if (typeof body === 'string') {
    // body is a string
} else {
    // not a string
}
```